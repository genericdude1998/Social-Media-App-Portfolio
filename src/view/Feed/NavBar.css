*{
    box-sizing: border-box;
    padding: 0px;
}

.baseNavBar{
    background-color: darkblue;
    height: 10vh;
    width: 100vw;
    min-width: 500px;
    line-height: 30px;
}

.baseNavBarContent{
    margin-left: 10px;
    width: 100%;
    display: flex;
    justify-content: flex-end;
}

.baseNavBarLink{
    color: white;
    text-decoration: none;
    margin-right: 50px;
}

.baseToggleThemeButton{
    margin-right: 20px;
    background-color: white;
    border: 1px solid gray;
}

.dark{
    background-color: darkred;
    color: white;
}

