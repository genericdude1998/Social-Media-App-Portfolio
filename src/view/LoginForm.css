*{
  box-sizing: border-box;
  padding: 0px; 
}

.baseForm{
    background-image: url('/assets/background-login.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    align-items:center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
}
.baseLabel{
    color: black;
    margin: 10px;
    text-align: center;
    font-weight: bold;
}
.baseInput{
   margin:5px;
   border: 0px;
   border: 1px solid black;

}
.baseButton{
    margin:20px;
    height: 20px;
    width: 50px;
    border: 1px solid black;
    background-color: white;
}

.light{

}

.dark{

}