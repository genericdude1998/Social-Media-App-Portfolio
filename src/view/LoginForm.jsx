import React from 'react';
import styles from './LoginForm.css'

const LoginForm = () => {
    const [username, setUsername] = React.useState('');
    const [password, setPassword] = React.useState('');

    return (
        <form className={styles.baseForm}>
            <label className={styles.baseLabel}>Username:</label>
            <input type='text' className={styles.baseInput}/>
            <label className={styles.baseLabel}>Password:</label>
            <input type='password'className={styles.baseInput}/>
            <button type='submit' className={styles.baseButton}>Login</button>
        </form>
    );
}

export default LoginForm;
