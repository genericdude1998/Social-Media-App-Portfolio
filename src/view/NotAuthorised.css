*{
    box-sizing: border-box;
    margin: 0px;
}
.notAuth{
    font-size: 3vw;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 10vw;
    height: 50vh;
    background-color: royalblue;
}

.link{
    text-decoration: none;
    font-weight: bold;
}