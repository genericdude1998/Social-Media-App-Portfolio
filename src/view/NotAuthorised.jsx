import React from 'react';
import { Link } from 'react-router-dom';
import styles from './NotAuthorised.css';

const NotAuthorised = () => {
    return (
        <div className={styles.notAuth}>
            You do not have the required authorisation.
            Please go back to <Link to={'/'} className={styles.link}>Login</Link>
        </div>
    );
}

export default NotAuthorised;
