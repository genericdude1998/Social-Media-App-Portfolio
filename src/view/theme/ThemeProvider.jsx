import React, { createContext } from 'react';
import { connect } from 'react-redux';
import { mapThemeDispatchToProps, mapThemeStateToProps } from '../mappers/mappers';
import PropTypes from 'prop-types';

export const ThemeContext = createContext();

const ThemeProvider = ({
    theme, onToggleTheme, children}) => 
    (
        <ThemeContext.Provider value={{onToggleTheme, theme}}>
            {children}
        </ThemeContext.Provider>
    );

ThemeProvider.propTypes = {
    theme: PropTypes.string,
    onToggleTheme: PropTypes.func,
    children: PropTypes.object,
}

const ConnectedThemeProvider = connect(mapThemeStateToProps,mapThemeDispatchToProps)(ThemeProvider);

export default ConnectedThemeProvider;
