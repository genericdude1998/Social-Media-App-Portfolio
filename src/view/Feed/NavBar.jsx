import React from 'react';
import { Link } from 'react-router-dom';
import UserLink from './UserLink';
import styles from './NavBar.css';
import { ThemeContext } from '../ThemeProvider';

const NavBar = () => {
    const {theme, onToggleTheme} = React.useContext(ThemeContext)
    return (
        <div className={`${styles.baseNavBar} ${theme === 'dark' ? styles.dark : null}`}>
            <div className={`${styles.baseNavBarContent} ${theme === 'dark' ? styles.dark : null}`}>
                <Link to='/feed' className={`${styles.baseNavBarLink} ${theme === 'dark' ? styles.dark : null}`}>Feed</Link>
                <Link to='/createPost' className={`${styles.baseNavBarLink} ${theme === 'dark' ? styles.dark : null}`}>Create New Post</Link>
                <UserLink id={1} name={'Me'} className={`${styles.baseNavBarLink} ${theme === 'dark' ? styles.dark : null}`}></UserLink>
                <button onClick={onToggleTheme}>Toggle Dark/Light Theme</button>
            </div>
        </div>
    );
}

export default NavBar;
