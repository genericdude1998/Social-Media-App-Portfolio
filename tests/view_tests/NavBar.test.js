import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext } from '../../src/view/ThemeProvider';
import NavBar from '../../src/view/Feed/NavBar';
import { mockTheme } from '../mockValues';

jest.spyOn(React, 'useContext').mockImplementation((context) => {
    return {theme:mockTheme, onToggleTheme: jest.fn()}
});
const NavBarWrapper = shallow(
    <NavBar/>
);

describe('NavBar', () => {
    it('should call useContext with expected params', () => {
        expect(React.useContext).toHaveBeenCalledWith(ThemeContext); 
    });
    it('should match Snapshot when theme is light', () => {
        jest.spyOn(React, 'useContext').mockImplementation((context) => {
            return {theme:'light', onToggleTheme: jest.fn()}
        });
        const NavBarWrapper = shallow(
            <NavBar/>
        );
        expect(NavBarWrapper).toMatchSnapshot();
    });
    it('should match Snapshot when theme is dark', () => {
        jest.spyOn(React, 'useContext').mockImplementation((context) => {
            return {theme:'dark', onToggleTheme: jest.fn()}
        });
        const NavBarWrapper = shallow(
            <NavBar/>
        );
        expect(NavBarWrapper).toMatchSnapshot();
    });
});
