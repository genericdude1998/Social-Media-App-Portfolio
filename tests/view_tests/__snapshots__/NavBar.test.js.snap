// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match Snapshot when theme is dark 1`] = `
<div
  className="baseNavBar dark"
>
  <div
    className="baseNavBarContent dark"
  >
    <Link
      className="baseNavBarLink dark"
      to="/feed"
    >
      Feed
    </Link>
    <Link
      className="baseNavBarLink dark"
      to="/createPost"
    >
      Create New Post
    </Link>
    <UserLink
      className="baseNavBarLink dark"
      id={1}
      name="Me"
    />
    <button
      onClick={[MockFunction]}
    >
      Toggle Dark/Light Theme
    </button>
  </div>
</div>
`;

exports[`NavBar should match Snapshot when theme is light 1`] = `
<div
  className="baseNavBar null"
>
  <div
    className="baseNavBarContent null"
  >
    <Link
      className="baseNavBarLink null"
      to="/feed"
    >
      Feed
    </Link>
    <Link
      className="baseNavBarLink null"
      to="/createPost"
    >
      Create New Post
    </Link>
    <UserLink
      className="baseNavBarLink null"
      id={1}
      name="Me"
    />
    <button
      onClick={[MockFunction]}
    >
      Toggle Dark/Light Theme
    </button>
  </div>
</div>
`;
